modules/dem_uart/test/test_dem_uart.py

```diff
@@ -0,0 +1,222 @@
+"""
+    test_dem_uart
+    
+    Cocotb-based unit test for UART-Device Emulation Module (DEM UART)
+"""
+import random
+import cocotb
+
+from cocotb.clock import Clock
+from cocotb.triggers import Timer, RisingEdge
+
+from osdtestlib.debug_interconnect import RegAccess, DiPacket, NocDiReader, NocDiWriter
+from osdtestlib.asserts import *
+from osdtestlib.exceptions import DebugEventFailedException
+
+# Debug Interconnect address for the DEM module
+MODULE_DI_ADDRESS = 1
+
+# Debug interconnect address for the testing module
+SENDER_DI_ADDRESS = 0
+
+
+@cocotb.coroutine
+def _init_dut(dut):
+    
+    #Setup clock
+    cocotb.fork(Clock(dut.clk, 1000).start())
+
+    # Reset DUT
+    dut._log.debug("Resetting DUT");
+    dut.rst <= 1
+    
+    dut.id <= MODULE_DI_ADDRESS
+    
+    for _ in range(2):
+        yield RisingEdge(dut.clk)
+    dut.rst <= 0    
+
+@cocotb.test()
+def test_dem_uart_baseregisters(dut):
+    """
+    Verify that the base configuration registers have the correct values.
+    """
+    
+    access = RegAccess(dut)
+    
+    yield _init_dut(dut)
+    
+    yield access.test_base_registers(MODULE_DI_ADDRESS, SENDER_DI_ADDRESS,
+                                     mod_vendor=1, mod_type=5, mod_version=0,
+                                     can_stall=True)
+
+@cocotb.coroutine
+def _activate_module(dut):
+    """
+    Set the MOD_CS_ACTIVE bit in the Control and Status register to 1 to
+    enable transmission of debug event packets.
+    """
+
+    access = RegAccess(dut)
+
+    yield access.write_register(dest=MODULE_DI_ADDRESS,
+                                src=SENDER_DI_ADDRESS,
+                                word_width=16,
+                                regaddr=DiPacket.BASE_REG.MOD_CS.value,
+                                value=1)
+
+@cocotb.test()
+def test_dem_uart_activation(dut):
+    """
+    Check if DEM_UART is handling the activation bit correctly
+    """
+
+    access = RegAccess(dut)
+
+    yield _init_dut(dut)
+
+    dut._log.info("Check contents of MOD_CS")
+    yield access.assert_reg_value(MODULE_DI_ADDRESS, SENDER_DI_ADDRESS,
+                                  DiPacket.BASE_REG.MOD_CS.value, 16, 0)
+
+    yield _activate_module(dut)
+
+    dut._log.info("Check contents of MOD_CS")
+    yield access.assert_reg_value(MODULE_DI_ADDRESS, SENDER_DI_ADDRESS,
+                                  DiPacket.BASE_REG.MOD_CS.value, 16, 1)
```

Comment:
This test is incomplete: it only tests if the activation bit is stored, but not if the (de-)activation actually does anything. Does the module stop sending data if it has been deactivated? Does it start to send again when it has been activated? This behavior should be added to the spec, and then tested for.

If you don't want to do it now that's fine. In this case remove the test case, open an issue describing the problem, and add it to your backlog at https://github.com/orgs/opensocdebug/projects/1